File: Leetcode11. Container with most water/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int maxVolume = 0;
        int L=0, R=n-1;
        while (L < R) {
            int w = R-L;
            int h=Math.min(height[L], height[R]);
            int area=w*h;
            maxVolume = Math.max(area, maxVolume);

            if (height[L] < height[R]) {
                L++;
            } else {
                R--;
            }
        }
        return maxVolume;
    }
}

File: Leetcode11. Container with most water/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_volume=0
        n=len(height)
        L=0 
        R=n-1
        while L < R:
            h = min(height[L], height[R])
            w=R-L
            area = w*h
            max_volume = max(area, max_volume)

            if height[L] < height[R]:
                L += 1
            else:
                R -= 1
        return max_volume
